nautobot/docs/installation/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,408 @@
+
+The old `constraints` field for a `Site`/`Region` related data model's (e.g. `Note`) `ObjectPermission` instance might look like this:
+
+```json
+{
+    "assigned_object_type": "dcim.site",
+    "assigned_object_id": "932d94ee-5571-40a0-903f-4274fcfbed32",
+    "assigned_object_id__in": ["932d94ee-5571-40a0-903f-4274fcfbed32", "e383db9a-dd55-464d-9e56-2f18bc03b32c"],
+}
+```
+
+To modify the data correctly, we need to:
+    1. Replace all occurrences of "dcim.site" and/or "dcim.region" with "dcim.location" in the **Value** portion (after ":") of the `assigned_object_type` Key.
+    2. Since Nautobot carries over the UUIDs of the old `Site`/`Region` instances when creating the new "Site"/"Region" type `Location` instances, we **do not** need to change the UUID values in the `assigned_object_id` and `assigned_object_id__in` Keys.
+
+The updated JSON data might look like this:
+
+```json
+{
+    "assigned_object_type": "dcim.location",
+    "assigned_object_id": "932d94ee-5571-40a0-903f-4274fcfbed32",
+    "assigned_object_id__in": ["932d94ee-5571-40a0-903f-4274fcfbed32", "e383db9a-dd55-464d-9e56-2f18bc03b32c"],
+}
+```
+
+## Region and Site Related Data Model Migration Guide For Existing Nautobot App installations
+
+In Nautobot 2.0.0, all the `Region` and `Site` related data models are being migrated to use `Location`. Below is a comprehensive guide for Nautobot App developers to migrate their `Region` and `Site` related data models to `Location`.
+
+We will be using `ExampleModel` as a relatively simple and hands-on example throughout this guide to better your understanding of the migration process.
+
+### Add Location Fields to Site and Region Related Data Models
+
+If the `ExampleModel` currently has a `site` ForeignKey field but it does not have a `location` ForeignKey field, you will need to add the `location` field before any other migrations in this guide.
+
+!!! note
+    You can skip this step only if your data models already have both a `site` (or `region`) field and a `location` field.
+
+```python
+# models.py
+
+class ExampleModel(OrganizationalModel):
+    site = models.ForeignKey(to="dcim.Site", on_delete=models.PROTECT, blank=True, null=True)
+    name = models.CharField(max_length=20, help_text="The name of this Example.")
+...
+```
+
+**DO NOT** delete the `site` ForeignKey field yet. As a first step, just add a `ForeignKey` to `dcim.Location` with all other arguments identical to the existing `dcim.Site` `ForeignKey`:
+
+```python
+# models.py
+
+class ExampleModel(OrganizationalModel):
+    site = models.ForeignKey(to="dcim.Site", on_delete=models.PROTECT, blank=True, null=True)
+    location = models.ForeignKey(to="dcim.Location", on_delete=models.PROTECT, blank=True, null=True)
+    name = models.CharField(max_length=20, help_text="The name of this Example.")
+...
+```
+
+Make the migration file by running `nautobot-server makemigrations [app_name] -n [migration_name]`, for example:
```

Comment:
What state do the Nautobot migrations need to be in at this point? Still a 1.5.x instance, or should 2.0 be already installed (but not yet migrated) at this point?